### ISSUE TITLE
Strategies working with CD servers?

### ISSUE THREAD

**Issue by PokeyTheCat** [vasiliyfomichev/SitecoreSitemapXML#14]
Does anyone have experience working with this module and CD servers? Is everyone just manually copying the sitemaps and robots.txt to their servers in that architecture setup? I'd love to hear how other people are handling it. In my case I have an internal CM server and load balanced CD servers.

**Comment by MythiMouse** [comment]
I've installed the source [Sitemap XML module](https://marketplace.sitecore.net/Modules/S/Sitemap_XML.aspx) today and found that I just had to change the below in the **SitemapXML.config** on the **CD** server:

from

`<event name="publish:end">`

to

`<event name="publish:end:remote">`

I'm thinking the forked module would work similarly? I think you'd need also just the **Sitemap.XML.dll** file on the CD server. After the above change, publishing on CM created the robots.txt and XML files on CD.

**Comment by PokeyTheCat** [comment]
I want to thank you for pointing this out. I had installed the module on my CD server but my initial testing didn't show anything working. 
Your comment led me verifying that the event handler was running. I kept looking at my logs on my CD server in my test environment and didn't see anything... so I wrote my own handler and deployed it to just log a message... long story short it was a permissions issue with my app pool on my CD server; it only had read only access to all of the directories. Once I fixed that my logging was working correctly AND my sitemaps were generating on my CD server correctly.

For anyone else that comes along, what I've done is deployed the module's DLL and config to my CD server. I adjusted the event that it would listen for to one listed above and I changed the database to web instead of master as my CD environments don't have master defined and don't have access to master.

Thank you @MythiMouse.
